server/src/main/scala/lsug/Meetup.scala

```diff
@@ -11,32 +14,47 @@ import cats.effect._
 
 import lsug.protocol.Meetup.{MeetupDotCom => P}
 import lsug.protocol.Link
+import scala.util.matching.Regex
+import cats.Monad
 
 object Meetup {
-
+  val eventMap: mutable.Map[String, (Int, LocalDateTime)] =
```

Comment:
Hey! Thanks for opening this PR; Meetup has been down frequently so it would be nice to have some cache for it :smile:.

There are a couple of points to consider:

1. It would probably be best if we extracted the logic into a separate class and passed it into the `apply` method. This would mean that it's easier to test the cache behavior as an independent unit
2. In general we avoid mutable state unless there's a good reason to have it (i.e. bottleneck for performance). We can use a [Ref](https://typelevel.org/cats-effect/concurrency/ref.html) with an _immutable_ map to get a primitive concurrent cache.
3. In general, `Map` doesn't provide any guarantees about it's behaviour under concurrent conditions; looking at the implementation of Scala's `Map`, I think it would be fine, but it's not great to rely on implementation details.

I think sketching out how the solution might be used first before implementing it is a good idea. Here's a starting point.

```scala
trait Cache[F[_], K, V] {
  def fetch(key: K, retrieve: K => F[V]): F[V]
}

object Cache {
  def apply[F[_]: Sync, K, V](evictionTime: Duration): Cache[F, K, V] = { 
    ???
  }
}
```

Thanks for joining the gitter chat and let me know if you encounter any issues.